src/Plugin/search_api/processor/AuthenticatedStatus.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\tide_authenticated_content\Plugin\search_api\processor;
+
+use Drupal\search_api\Plugin\PluginFormTrait;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Excludes authenticated content from node indexes.
+ *
+ * @SearchApiProcessor(
+ *   id = "authenticated_status",
+ *   label = @Translation("Exclude authenticated status"),
+ *   description = @Translation("Remove authenticated content from being indexed."),
+ *   stages = {
+ *     "alter_items" = -1,
+ *   },
+ * )
+ */
+class AuthenticatedStatus extends ProcessorPluginBase {
+
+  use PluginFormTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    /** @var static $processor */
+    $processor = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+
+    return $processor;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alterIndexedItems(array &$items) {
+
+    /** @var \Drupal\search_api\Item\ItemInterface $item */
+    foreach ($items as $item_id => $item) {
+      $object = $item->getOriginalObject()->getValue();
+      $bundle = $object->bundle();
+
+      if ($bundle === 'landing_page' && $object->hasField('field_authenticated_content')) {
+        $value = $object->get('field_authenticated_content')->getValue();
+        if ($value[0]['target_id']) {
```

Comment:
```suggestion
if (!$object->field_authenticated_content->isEmpty()) {
```
You don't need to check the value, just check if the field is empty or not. Just something like above.

Reply:
@vincent-gao just wondering, the drop down has a "none" value as well. So if it is selected as "none" does it count as the field is empty as well ? 
![Screen Shot 2022-08-04 at 12 04 51 pm](https://user-images.githubusercontent.com/20810541/182747404-bc1d8228-aec2-4cd0-a45d-27a286d52db4.png)
@yeniatencio it might be worth to check if you select "none" in the drop down, does it consider the field to be empty or not.

Reply:
@MdNadimHossain , Yes I have tested Vincent suggestion and it is working as expected.